datafusion/sqllogictest/test_files/case.slt

```diff
@@ -50,7 +50,7 @@ NULL
 6
 NULL
 NULL
-7
```

Comment:
this last row has `a IS NULL`,`b = 7`

So the idea is that a > 2 is not true when `a IS NULL` so should fall through to the `ELSE null` clause